src/Avalonia.OpenGL/OpenGlException.cs

```diff
@@ -1,12 +1,50 @@
 using System;
+using System.Reflection;
 
 namespace Avalonia.OpenGL
 {
     public class OpenGlException : Exception
     {
+        public int? ErrorCode { get; private set; }
+
         public OpenGlException(string message) : base(message)
         {
+        }
+
+        private OpenGlException(string message, int errorCode) : base(message)
+        {
+            ErrorCode = errorCode;
+        }
+
+        public static OpenGlException GetFormattedException(string funcName, EglInterface egl)
+        {
+            return GetFormattedException(typeof(EglConsts).GetFields(), funcName, 0x3000, 0x301F, egl.GetError());
+        }
+
+        public static OpenGlException GetFormattedException(string funcName, GlInterface gl)
+        {
+            return GetFormattedException(typeof(GlConsts).GetFields(), funcName, 0x0500, 0x0505, gl.GetError());
+        }
+
+        private static OpenGlException GetFormattedException(
+            FieldInfo[] fields, string funcName, int minValue, int maxValue, int errorCode)
+        {
+            foreach (var field in fields)
```

Comment:
`Enum.GetValues(typeof(EglConsts)).Cast<int>().Where(code => code >= minValue && code <= maxValue).FirstOrDefault();`

Reply:
EglConsts and GlConsts are not enumerations, but regular classes with const fields.

Reply:
I'm talking about a separate enumeration for errors.
```cs
enum GlErrors
{
    GL_INVALID_ENUM = 0x0500,
    GL_INVALID_VALUE = 0x0501,
    GL_INVALID_OPERATION = 0x0502,
    GL_STACK_OVERFLOW = 0x0503,
    GL_STACK_UNDERFLOW = 0x0504,
    GL_OUT_OF_MEMORY = 0x0505
}

enum EglErrors
{
    EGL_BAD_ACCESS = 0x3002,
    EGL_BAD_ALLOC = 0x3003,
    EGL_BAD_ATTRIBUTE = 0x3004,
    EGL_BAD_CONFIG = 0x3005,
    EGL_BAD_CONTEXT = 0x3006,
    EGL_BAD_CURRENT_SURFACE = 0x3007,
    EGL_BAD_DISPLAY = 0x3008,
    EGL_BAD_MATCH = 0x3009,
    EGL_BAD_NATIVE_PIXMAP = 0x300A,
    EGL_BAD_NATIVE_WINDOW = 0x300B,
    EGL_BAD_PARAMETER = 0x300C,
    EGL_BAD_SURFACE = 0x300D
}
```